solr/core/src/java/org/apache/solr/search/SolrDocumentFetcher.java

```diff
@@ -114,11 +114,33 @@ public class SolrDocumentFetcher {
 
   private final Set<String> largeFields;
 
-  private Collection<String> storedHighlightFieldNames; // lazy populated; use getter
+  private final Collection<String>[] storedHighlightFieldNames; // lazy populated; use getter
+
+  private final Collection<String>[] indexedFieldNames; // lazy populated; use getter
+
+  private final StoredFields storedFields;
+
+  private SolrDocumentFetcher(SolrDocumentFetcher template, StoredFields storedFields) {
+    this.searcher = template.searcher;
+    this.nLeaves = template.nLeaves;
+    this.enableLazyFieldLoading = template.enableLazyFieldLoading;
+    this.documentCache = template.documentCache;
+    this.nonStoredDVsUsedAsStored = template.nonStoredDVsUsedAsStored;
+    this.allNonStoredDVs = template.allNonStoredDVs;
+    this.nonStoredDVsWithoutCopyTargets = template.nonStoredDVsWithoutCopyTargets;
+    this.largeFields = template.largeFields;
+    this.dvsCanSubstituteStored = template.dvsCanSubstituteStored;
+    this.allStored = template.allStored;
+    this.storedHighlightFieldNames = template.indexedFieldNames;
+    this.indexedFieldNames = template.indexedFieldNames;
+    this.storedFields = storedFields;
+  }
 
-  private Collection<String> indexedFieldNames; // lazy populated; use getter
+  public SolrDocumentFetcher createInstance() throws IOException {
```

Comment:
This is a weird method; and there's no javadoc.  A class that has a "createInstance" that returns a type of itself?  "clone" would be a better name.  Let's not make this public so people can't even easily call it; only SolrIndexSearcher should call it.

Reply:
Updated to `clone()` which should be self-explanatory.